external-crates/move/crates/move-command-line-common/src/parser.rs

```diff
@@ -163,17 +190,11 @@ impl<'a, I: Iterator<Item = (TypeToken, &'a str)>> Parser<'a, TypeToken, I> {
                 self.advance(TypeToken::Gt)?;
                 ParsedType::Vector(Box::new(ty))
             }
-            (TypeToken::Ident, _) | (TypeToken::AddressIdent, _) => {
-                let addr_tok = match tok {
-                    TypeToken::Ident => ValueToken::Ident,
-                    TypeToken::AddressIdent => ValueToken::Number,
-                    _ => unreachable!(),
-                };
-                let address = parse_address_impl(addr_tok, contents)?;
-                self.advance(TypeToken::ColonColon)?;
-                let module_contents = self.advance(TypeToken::Ident)?;
+
+            (tok @ (TypeToken::Ident | TypeToken::AddressIdent), contents) => {
+                let module = self.parse_module_id_impl(tok, contents)?;
                 self.advance(TypeToken::ColonColon)?;
-                let struct_contents = self.advance(TypeToken::Ident)?;
+                let name = self.advance(TypeToken::Ident)?.to_owned();
```

Comment:
so we've parsed the address and module, and now looking at the Type?

Reply:
That's right